icu4c/source/test/cintltst/ucptrietest.c

```diff
@@ -705,6 +706,59 @@ testBuilder(const char *testName, const UMutableCPTrie *mutableTrie,
     testTrieGetRanges(testName, NULL, mutableTrie, UCPMAP_RANGE_NORMAL, 0, checkRanges, countCheckRanges);
 }
 
+static void
+trieTestGolden(const char *testName,
+             const UCPTrie* trie,
+             const CheckRange checkRanges[],
+             int32_t countCheckRanges) {
+    log_verbose("golden testing Trie '%s'\n", testName);
+
+    UErrorCode status = U_ZERO_ERROR;
+    const char *testdatapath = loadSourceTestData(&status);
+    char goldendatapath[512];
+    // note: snprintf always writes a NUL terminator.
+    snprintf(goldendatapath, 512, "%sucptrie%s%s.toml",
+        testdatapath, U_FILE_SEP_STRING, testName);
+
+    // Write the data into a tmpfile (memstream is not portable)
+    FILE* stream = tmpfile();
+    fputs("[ucptrie.struct]\n", stream);
```

Comment:
Add copyright header

Reply:
@markusicu Is there a function somewhere that will add the copyright header automatically, or should I hard-code the copyright header into the file writer here?

Reply:
We have https://github.com/unicode-org/icu/blob/master/icu4c/source/tools/toolutil/writesrc.cpp
- `usrc_create()` writes a C-style header with `//` comments
- `usrc_createTextData()` writes a text/script/Python-style header with `#` comments; and has a TODO to add a parameter like `usrc_create()` has for the shorter copyright header after 2016

Reply:
I refactored things a bit in writesrc.cpp and added a separate function for adding the copyright header without creating a new file. PTAL.